deepchecks/vision/checks/performance/robustness_report.py

```diff
@@ -0,0 +1,406 @@
+                           title=dict(text='Performance Comparison', font=dict(size=20)),
+                           showlegend=False)
+         .update_xaxes(title=None, type='category', tickangle=30))
+        return fig
+
+    def _create_top_affected_graph(self, top_affected_dict):
+        metrics = sorted(top_affected_dict.keys())
+        fig = make_subplots(rows=1, cols=len(metrics), subplot_titles=metrics)
+
+        for index, metric in enumerate(metrics):
+            metric_classes = top_affected_dict[metric]
+            # Take the worst affected classes
+            x = []
+            y = []
+            custom_data = []
+            for class_info in metric_classes:
+                x.append(class_info['class'])
+                y.append(class_info['value'])
+                custom_data.append([format_percent(class_info['diff']), class_info['samples']])
+
+            # Plotly have a bug that if all y values are zero text position 'auto' doesn't work
+            textposition = 'outside' if sum(y) == 0 else 'auto'
+            fig.add_trace(go.Bar(name=metric, x=x, y=y, customdata=custom_data, texttemplate='%{customdata[0]}',
+                                 textposition=textposition, hovertemplate='Number of samples: %{customdata[1]}'),
+                          row=1, col=index + 1)
+
+        (fig.update_layout(font=dict(size=12), height=300, width=600 * len(metrics),
+                           title=dict(text='Top Affected Classes', font=dict(size=20)),
+                           showlegend=False)
+         .update_xaxes(title=None, type='category', tickangle=30, tickprefix='Class ', automargin=True)
+         .update_yaxes(automargin=True))
+
+        return fig
+
+
+def augmentation_name(aug):
+    if isinstance(aug, imgaug.augmenters.Augmenter):
+        return aug.name
+    elif isinstance(aug, albumentations.BasicTransform):
+        return aug.get_class_fullname()
+    else:
+        raise DeepchecksValueError(f'Unsupported augmentation type {type(aug)}')
+
+
+def get_random_image_pairs_from_dataset(original_dataset: VisionData,
+                                        augmented_dataset: VisionData,
+                                        top_affected_classes: dict):
+    """Get image pairs from 2 datasets.
+
+    We iterate the internal dataset object directly to avoid randomness
+    Dataset returns data points as processed images, making this currently not really usable
+    To avoid making more assumptions this currently stays as-is
+    Note that images return in RGB format, ond to visualize them using OpenCV the final dimension should be
+    transposed;
+    can be done via image = image[:, :, ::-1] or cv2.cvtColor(img, cv2.COLOR_RGB2BGR)
+    """
+    classes_to_show = {
+        class_info['class']: class_info['diff']
+        for classes_list in top_affected_classes.values()
+        for class_info in classes_list
+    }
+
+    baseline_sampler = iter(original_dataset.get_data_loader().dataset)
```

Comment:
Maybe we should check that iter on two copies of the original returns the same image, before we rely on this here to produce the pairs.

Reply:
I think it doesn't matter. if it doesn't you will still want to show examples, so the only difference might be some more space between original and augmented examples or something which is not that critical